### ISSUE TITLE
Clarify in documentation that passwords are hashed, not encrypted.

### ISSUE THREAD

**Issue by joelatschool** [mikeGEINE/FPauth#3]
Sorry if this sounds like a nitpick but for me it seemed like a huge red flag when I first saw it: 
https://github.com/mikeGEINE/FPauth/blob/0eaef19a14e62910cbd9fbafa07105e4989c6e2e/src/strategies/password.ml#L11

It's really important if you're handling authentication to understand that [passwords must be **hashed**, not **encrypted**](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#hashing-vs-encryption). If I understand this package correctly, it's not actually handling password storage, just password verification, and the `Argon2.ID.verify` function correctly verifies against a hashed password and not an encrypted password, so I think this is just something misnamed rather than an actual security failure. Still, I worry that it has the potential to misinform users of the library about how they should be securely handling passwords, and makes the library look insecure and dangerous.

I'd recommend renaming `encrypted_password` to `hashed_password` or `encoded_password` (to match the terminology used in [the Argon2](https://khady.github.io/ocaml-argon2/argon2/index.html) documentation).

**Comment by mikeGEINE** [comment]
Yeah, you are right. Passwords are meant to be hashed, of course. However, I used Devise gem for Ruby on Rails as a reference, which adds an `encrypted_password` field to User model:

https://github.com/heartcombo/devise/blob/e2242a95f3bb2e68ec0e9a064238ff7af6429545/lib/devise/models/database_authenticatable.rb#L67C8

I'll think about this, maybe it really should be renamed. Thanks! :)

**Comment by joelatschool** [comment]
Got it, that's definitely fair. Sorry for freaking out about it then, it's just that without that context it really looked like a red flag. It's probably fine to just add something to the documentation explaining that despite the name it should be a hashed password, and maybe mention that it's named to match Devise, just like Devise does in its documentation now.
